### PR TITLE
`Assessment`: Fix export button in scores page

### PR DESCRIPTION
**Pull request by az108**
### Motivation and Context
Analyzing Student Performance and Results is important after Exams or Exercises. This helps instructors decide if the error is from Students side or Artemis side.
Therefore I have fixed the following Issue which was not allowing the Export to a CSV file.
https://github.com/ls1intum/Artemis/issues/8422

### Description
I added a null check to make sure there is no call on a undefined attribute on the client side.

### Steps for Testing

Prerequisites:
- 1 Instructor
- 2 Students
- 1 Programming Exercise with Complaints enabled

1. Log in to Artemis
2. Navigate to Course Administration
3. Navigate to a Programming Exericse
4. Navigate to Scores
5. Press the Export Button and receive a CSV file

#### Performance Review
- [ ] I (as a reviewer) confirm that the client changes (in particular related to REST calls and UI responsiveness) are implemented with a very good performance 
#### Code Review
- [x] Code Review 1
- [ ] Code Review 2
#### Manual Tests
- [x] Test 1
- [x] Test 2

## Summary by CodeRabbit

- **Refactor**
	- Simplified the data structure and interactions related to grading criteria in exercises by removing the handling of `pointsPerCriterion` across various components and services. This change streamlines the process of managing exercise scores and results, enhancing the user experience by focusing on essential data.

### CODE REVIEW

File: src/main/webapp/app/exercises/shared/result/result.service.ts [hunk 1]
```diff
@@ -217,9 +217,11 @@ export class ResultService implements IResultService {
             res.body.forEach((resultWithPoints: ResultWithPointsPerGradingCriterion) => {
                 this.convertResultDatesFromServer(resultWithPoints.result);
                 const pointsMap = new Map<number, number>();
-                Object.keys(resultWithPoints.pointsPerCriterion).forEach((key) => {
-                    pointsMap.set(Number(key), resultWithPoints.pointsPerCriterion[key]);
-                });
+                if (resultWithPoints.pointsPerCriterion != undefined) {
```
Comment (Strohgelaender) [inline src/main/webapp/app/exercises/shared/result/result.service.ts:220]:
```suggestion
                if (resultWithPoints.pointsPerCriterion) {
```

**Review by sarpsahinalp — APPROVED** [review]
Tested during testing session, lgtm

**Review by egekurt123 — APPROVED** [review]
Tested manually on ts2, works as expected

**Review by coolchock — APPROVED** [review]
lgtm

**Review by florian-glombik — APPROVED** [review]
Demonstrated in testing session, the button works again

`Export Results` does not include users without of a submission even though they are displayed on the scores table, however - I guess that this is not a problem since students who did not start the exercise wont be displayed in the scores view aswell.

It could only be confusing that `Export Names` does not ignore the users without submission and therefore without of a result.

**Pull request merged**